storage/src/vespa/storage/persistence/apply_bucket_diff_entry_result.h

```diff
@@ -0,0 +1,30 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include <vespa/document/base/documentid.h>
+#include <vespa/persistence/spi/bucket.h>
+#include <future>
+
+namespace storage::spi { class Result; }
+
+namespace storage {
+
+/*
+ * Result of a bucket diff entry spi operation (putAsync or removeAsync)
+ */
+class ApplyBucketDiffEntryResult {
+    using FutureResult = std::future<std::unique_ptr<spi::Result>>;
+    FutureResult         _future_result;
+    spi::Bucket          _bucket;
+    document::DocumentId _doc_id;
+    const char*          _op;
+
+public:
+    ApplyBucketDiffEntryResult(FutureResult future_result, spi::Bucket bucket, document::DocumentId doc_id, const char *op);
+    ApplyBucketDiffEntryResult(ApplyBucketDiffEntryResult &&rhs);
+    ~ApplyBucketDiffEntryResult();
+    void check_result();
```

Comment:
Consider calling this something like `wait_and_check_result()` to indicate it also waits for the result to be available (or split into two functions)